File: lib/models/aluno.dart
class Aluno {
  final int id;
  final String nome;
  final DateTime dataNascimento;
  final int idade;
  final String? municipio;
  final String? sexo;
  final String? estado;
  final String? nomePai;
  final String? nomeMae;
  final String? cpf;
  final String? telefoneResponsavel;
  final String? observacoes;
  final bool ativo;

  Aluno({
    required this.id,
    required this.nome,
    required this.dataNascimento,
    required this.idade,
    this.nomePai,
    this.municipio,
    this.sexo,
    this.estado,
    this.nomeMae,
    this.cpf,
    this.telefoneResponsavel,
    this.observacoes,
    required this.ativo,
  });

  factory Aluno.fromJson(Map<String, dynamic> json) {
    return Aluno(
      id: json['id'],
      nome: json['nome'],
      dataNascimento: DateTime.parse(json['dataNascimento'].toString()),
      idade: json['idade'],
      municipio: json['municipio'],
      sexo: json['sexo'],
      estado: json['estado'],
      nomePai: json['nomePai'],
      nomeMae: json['nomeMae'],
      cpf: json['cpf'],
      telefoneResponsavel: json['telefoneResponsavel'],
      observacoes: json['observacoes'],
      ativo: json['ativo'],
    );
  }

  String get dataNascimentoFormatada {
    return '${dataNascimento.day.toString().padLeft(2, '0')}/'
        '${dataNascimento.month.toString().padLeft(2, '0')}/'
        '${dataNascimento.year}';
  }
}

File: backend/Models/Aluno.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace NeuroGest.API.Models;

[Table("alunos")]
public class Aluno
{
    [Key]
    [Column("id")]
    public int Id { get; set; }

    [Required]
    [MaxLength(150)]
    [Column("nome")]
    public string Nome { get; set; } = string.Empty;

    [Required]
    [Column("data_nascimento")]
    public DateOnly DataNascimento { get; set; }

    // Calculado dinamicamente — não salvo no banco
    [NotMapped]
    public int Idade => CalcularIdade(DataNascimento);

    [MaxLength(150)]
    [Column("nome_pai")]
    public string? NomePai { get; set; }

    [Required]
    [MaxLength(150)]
    [Column("nome_mae")]
    public string NomeMae { get; set; } = string.Empty;

    [Required]
    [MaxLength(14)]
    [Column("cpf")]
    public string Cpf { get; set; } = string.Empty;

    [MaxLength(14)]
    [Column("cpf_pai")]
    public string? CpfPai { get; set; }

    [MaxLength(14)]
    [Column("cpf_mae")]
    public string? CpfMae { get; set; }

    [MaxLength(20)]
    [Column("telefone_responsavel")]
    public string? TelefoneResponsavel { get; set; }

    [MaxLength(200)]
    [Column("observacoes")]
    public string? Observacoes { get; set; }

    [Column("ativo")]
    public bool Ativo { get; set; } = true;

    [Column("criado_em")]
    public DateTime CriadoEm { get; set; } = DateTime.UtcNow;

    private static int CalcularIdade(DateOnly nascimento)
    {
        var hoje = DateOnly.FromDateTime(DateTime.Today);
        var idade = hoje.Year - nascimento.Year;
        if (nascimento > hoje.AddYears(-idade)) idade--;
        return idade;
    }
}
